torchvision/prototype/datasets/_builtin/celeba.py

```diff
@@ -0,0 +1,200 @@
+import csv
+import io
+import pathlib
+from typing import Any, Callable, Dict, List, Optional, Tuple, Sequence
+
+import torch
+from torch.utils.data import IterDataPipe
+from torch.utils.data.datapipes.iter import (
+    Mapper,
+    Shuffler,
+    Filter,
+    ZipArchiveReader,
+)
+from torchdata.datapipes.iter import KeyZipper
+from torchvision.prototype.datasets.utils import (
+    Dataset,
+    DatasetConfig,
+    DatasetInfo,
+    GDriveResource,
+    OnlineResource,
+)
+from torchvision.prototype.datasets.utils._internal import INFINITE_BUFFER_SIZE
+
+
+class CelebACSVParser(IterDataPipe):
+    def __init__(
+        self,
+        datapipe,
+        *,
+        has_header,
+    ):
+        self.datapipe = datapipe
+        self.has_header = has_header
+        self._fmtparams = dict(delimiter=" ", skipinitialspace=True)
+
+    def __iter__(self):
+        for _, file in self.datapipe:
+            file = (line.decode() for line in file)
+
+            if self.has_header:
+                # The first row is skipped, because it only contains the number of samples
+                next(file)
+
+                # Empty field names are filtered out, because some files have an extr white space after the header
+                # line, which is recognized as extra column
+                fieldnames = [name for name in next(csv.reader([next(file)], **self._fmtparams)) if name]
+                # Some files do not include a label for the image ID column
+                if fieldnames[0] != "image_id":
+                    fieldnames.insert(0, "image_id")
```

Comment:
This is super annoying that we can't use `DictReader`.
Since we have three datapipes with header, could we hard-code the `fieldnames` for each one and use `CSVDictReader(dp, skip_lines=2, fieldnames=[...])`

Reply:
We could, but there is more to it. Note that we also need to `map` the output so that it is a tuple with the image id first and the remaining row second. At this point we would probably have a more elaborate implementation bending everything to use the default building blocks than to just write our own. 

I agree it is annoying, but writing and understanding this custom parser is not hard so I feel its warranted.